### ISSUE TITLE
can you give a example for Contrastive Learning?

### ISSUE THREAD

**Issue by ScottishFold007** [jamescalam/transformers#3]
Now that comparative learning is hotly researched and used in NLP, can you do an in-depth analysis on ContrastiveLoss, TripletLoss and Multiple Negatives Ranking Loss, etc., without resorting to existing sentence-transformers?

**Comment by jamescalam** [comment]
TLDR - yes, but it will take a couple of months

Sorry for the late response, the current goal of the repo is to create a pretty broad (and shallow) overview of transformers for NLP.

To me, comparative learning belongs within that goal only as a high-level summary eg to the extent covered in the similarity section.

However, I'm planning to take one of two routes after that first goal is complete. Both routes cover diving deeper into specific topics (and comparative learning is a priority - maybe even number 1), the first would be to cover these in independent repos, and the second to simply cover them in this repo.

I'm leaning towards the latter, so yes, I almost definitely will be covering that, but it will take some time.

**Comment by ScottishFold007** [comment]
> TLDR - yes, but it will take a couple of months
> 
> Sorry for the late response, the current goal of the repo is to create a pretty broad (and shallow) overview of transformers for NLP.
> 
> To me, comparative learning belongs within that goal only as a high-level summary eg to the extent covered in the similarity section.
> 
> However, I'm planning to take one of two routes after that first goal is complete. Both routes cover diving deeper into specific topics (and comparative learning is a priority - maybe even number 1), the first would be to cover these in independent repos, and the second to simply cover them in this repo.
> 
> I'm leaning towards the latter, so yes, I almost definitely will be covering that, but it will take some time.

Good. Looking forward to your masterpiece!

**Comment by jamescalam** [comment]
Closing the issue as I've *finally* begun to cover [contrastive learning over on YouTube](https://www.youtube.com/watch?v=or5ew7dqA-c&list=PLIUOU7oqGTLgz-BI8bNMVGwQxIMuQddJO&index=4)
